packages/quantic/force-app/main/default/lwc/quanticModal/quanticModal.js

```diff
@@ -60,6 +64,11 @@ export default class QuanticModal extends LightningElement {
    */
   @api openModal() {
     this.isVisible = true;
+    // 500ms is the duration time of the animation to open the modal.
+    // eslint-disable-next-line @lwc/lwc/no-async-operation
+    setTimeout(() => {
+      this.focusOnFirstElement();
+    }, 500);
```

Comment:
yeah, actually the focus can be set even before the transition is over.
Basically this animation consists of changing the CSS visibility from `hidden` to `visible` and the CSS opacity from `0` to `1`.
The transition from opacity `0` to `1` will take 500ms while the transition from visibility `hidden` to `visible` will happen instantly because the visibility values are not numeric.
At 0ms the visibility is `hidden`, so we can’t yet set the focus on the modal because a hidden element is not accessible, but when the animation is started this visibility value will change instantly to `visible`  which makes the elements inside the modal accessible and focusable.
What I wanna say is that I could have set the timeout to only 1ms and this will still work properly cause at 1ms the CSS visibility is already set to `visible`.
So yeah, a timeout of 500ms is a full proof that this will execute after the end of the transition that matters for setting the focus.